src/logics/balancer-v2/logic.flash-loan.ts

```diff
@@ -30,6 +44,32 @@ export class FlashLoanLogic extends core.Logic implements core.LogicTokenListInt
     return tokenList;
   }
 
+  async quote(params: FlashLoanLogicParams) {
+    const { outputs: loans } = params;
+
+    const vaultAddress = getContractAddress(this.chainId, 'Vault');
+    const calls: common.Multicall2.CallStruct[] = loans.map((loan) => ({
+      target: loan.token.address,
+      callData: this.erc20Iface.encodeFunctionData('balanceOf', [vaultAddress]),
+    }));
+    const { returnData } = await this.multicall2.callStatic.aggregate(calls);
+
+    const repays = new common.TokenAmounts();
+    const fees = new common.TokenAmounts();
+    for (let i = 0; i < loans.length; i++) {
+      const loan = loans.at(i);
+      const [balance] = this.erc20Iface.decodeFunctionResult('balanceOf', returnData[i]);
+      const avaliableToBorrow = new common.TokenAmount(loan.token).setWei(balance);
+      invariant(avaliableToBorrow.gte(loan), `insufficient borrowing capacity for the asset: ${loan.token.address}`);
```

Comment:
Currently, the logics repo does not validate the input data's correctness. It is expected that the validation and control will be done at the API layer.
@zodahu , what do you think about this?

Reply:
Benefit: If the client does bring in two identical tokens, it can correctly calculate the available amount (but it will be fail on-chain).
Drawback: It unnecessarily adds more code, because the Balancer Vault contract does not allow this. If the client intentionally does this, it only causes trouble for themselves.

Thus I think it's unnecessary to check.

Another point is, it will also need to be added to the unit tests, making this scenario appear as if it is legitimate and expected, but actually it's not.

Reply:
@zodahu About `it will also need to be added to the unit tests, making this scenario appear as if it is legitimate and expected, but actually it's not.`
I think that it would be better to add validation rather than writing unit tests that simply indicate that the contract is not working as expected. However, we should allow such cases.

Reply:
Checking for duplicates within the loans is good.

Reply:
added validation